platform.js

```diff
@@ -423,5 +423,52 @@ module.exports = {
         });
       }
     });
+  },
+  runFunction(func, retryCount, retryTimeout, retryDelay) {
+    if(typeof(func) !== "function") {
+      return Promise.reject(["func should be a function"]);
+    }
+
+    return new Promise((resolve, reject)=>{
+      var errors = [];
+
+      internalRun(retryCount);
+
+      function internalRun(retries) {
+        var timer;
+
+        if(retryTimeout) {
+          timer = setTimeout(()=>{
+            handleError("function call timed out");
+          }, retryTimeout);
+        }
+
+        func()
+          .then(clearTimer)
```

Comment:
So func has to be a function but it also has to be a function that returns a promise. We should guard for that, or maybe coerce to a Promise via `Promise.resolve(func()).then...`